releases/fuji/compose-files/docker-compose-fuji-1.0.0.yml

```diff
@@ -0,0 +1,579 @@
+#  * Copyright 2019 Intel Corporation
+#  *
+#  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+#  * in compliance with the License. You may obtain a copy of the License at
+#  *
+#  * http://www.apache.org/licenses/LICENSE-2.0
+#  *
+#  * Unless required by applicable law or agreed to in writing, software distributed under the License
+#  * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  * or implied. See the License for the specific language governing permissions and limitations under
+#  * the License.
+#  *
+#  * @author: Jim White, Dell
+#  * added: July 26, 2019
+#  * @author: Jim Wang, Intel Corp
+#  * updated: October 8, 2019
+#  *******************************************************************************/
+
+version: '3'
+volumes:
+  db-data:
+  log-data:
+  consul-config:
+  consul-data:
+  portainer_data:
+  vault-config:
+  vault-file:
+  vault-logs:
+  secrets-setup-cache:
+
+services:
+  volume:
+    image: edgexfoundry/docker-edgex-volume:1.0.0
+    container_name: edgex-files
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-files
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+      - secrets-setup-cache:/etc/edgex/pki
+
+  consul:
+    image: consul:1.3.1
+    ports:
+      - "8400:8400"
+      - "8500:8500"
+      - "8600:8600"
+    container_name: edgex-core-consul
+    hostname: edgex-core-consul
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-consul
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+    depends_on:
+      - volume
+
+  config-seed:
+    image: edgexfoundry/docker-core-config-seed-go:1.0.0
+    container_name: edgex-config-seed
+    hostname: edgex-core-config-seed
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-config-seed
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+    depends_on:
+      - volume
+      - consul
+
+  vault:
+    image: edgexfoundry/docker-edgex-vault:1.1.0-dev
+    container_name: edgex-vault
+    hostname: edgex-vault
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-vault
+    ports:
+      - "8200:8200"
+    cap_add:
+      - "IPC_LOCK"
+    environment:
+      - VAULT_ADDR=https://edgex-vault:8200
+      - |
```

Comment:
this is kind of weird to read putting multi-line hcl file environment variable inside a YAML file. 
Can we just put this inside a file using a here-doc in the `command` section to write this to that file? We specify the config file in the snap when we run vault easily enough

Reply:
I think we can if I know how to do that. Can you give us some pointers on how to do that?

Reply:
to provide the location of the file, you just do something like:

https://github.com/edgexfoundry/edgex-go/blob/76aa1b5089da969faeb237ca84695ae805c4b411/snap/snapcraft.yaml#L128

and to use a here-doc to write the file, do something like:

```bash
cat <<'EOF'
first line
second line

line after an empty line
EOF > config.hcl
```

this writes everything in between `'EOF'` and `EOF` (not including `EOF`) to config.hcl, including the empty line.

Reply:
> 
> 
> to provide the location of the file, you just do something like:
> 
> https://github.com/edgexfoundry/edgex-go/blob/76aa1b5089da969faeb237ca84695ae805c4b411/snap/snapcraft.yaml#L128
> 
> and to use a here-doc to write the file, do something like:
> 
> ```shell
> cat <<'EOF'
> first line
> second line
> 
> line after an empty line
> EOF > config.hcl
> ```
> 
> this writes everything in between `'EOF'` and `EOF` (not including `EOF`) to config.hcl, including the empty line.

@anonymouse64 Ian, I tried it and this way actually makes the docker-compose `command` section very verbose with these extra `cat<<EOF....`  shell script.  Other way is to provide the pre-defined file inside the docker file but I do not like that solution either since it is kind of hard-coded all configuration stuff into docker image.  So I think the middle spot between two is the current implementation as it is.